Author: Otto Kekäläinen <otto@debian.org>
Date:   Fri Jul 8 23:52:23 2016 +0300
  Rename libmysqlclient to libmariadbclient in order to ship a
  libmariadbclient.so.18 in libmariadbclient18 and
  libmariadbclient.so in libmariadbclient-dev. There is a separate
  libmariadbclient-dev-compat package that links libmysqlclient.so
  to libmariadbclient.so.

--- mariadb-10.1.orig/client/CMakeLists.txt
+++ mariadb-10.1/client/CMakeLists.txt
@@ -32,7 +32,7 @@ COPY_OPENSSL_DLLS(copy_openssl_client)
 ADD_DEFINITIONS(${SSL_DEFINES})
 MYSQL_ADD_EXECUTABLE(mysql completion_hash.cc mysql.cc readline.cc
                            ${CMAKE_SOURCE_DIR}/sql/sql_string.cc)
-TARGET_LINK_LIBRARIES(mysql mysqlclient)
+TARGET_LINK_LIBRARIES(mysql mariadbclient)
 IF(UNIX)
   TARGET_LINK_LIBRARIES(mysql ${MY_READLINE_LIBRARY})
   SET_TARGET_PROPERTIES(mysql PROPERTIES ENABLE_EXPORTS TRUE)
@@ -40,39 +40,39 @@ ENDIF(UNIX)
 
 MYSQL_ADD_EXECUTABLE(mysqltest mysqltest.cc COMPONENT Test)
 SET_SOURCE_FILES_PROPERTIES(mysqltest.cc PROPERTIES COMPILE_FLAGS "-DTHREADS")
-TARGET_LINK_LIBRARIES(mysqltest mysqlclient pcre pcreposix)
+TARGET_LINK_LIBRARIES(mysqltest mariadbclient pcre pcreposix)
 SET_TARGET_PROPERTIES(mysqltest PROPERTIES ENABLE_EXPORTS TRUE)
 
 
 MYSQL_ADD_EXECUTABLE(mysqlcheck mysqlcheck.c)
-TARGET_LINK_LIBRARIES(mysqlcheck mysqlclient)
+TARGET_LINK_LIBRARIES(mysqlcheck mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysqldump mysqldump.c ../sql-common/my_user.c)
-TARGET_LINK_LIBRARIES(mysqldump mysqlclient)
+TARGET_LINK_LIBRARIES(mysqldump mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysqlimport mysqlimport.c)
 SET_SOURCE_FILES_PROPERTIES(mysqlimport.c PROPERTIES COMPILE_FLAGS "-DTHREADS")
-TARGET_LINK_LIBRARIES(mysqlimport mysqlclient)
+TARGET_LINK_LIBRARIES(mysqlimport mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysql_upgrade mysql_upgrade.c COMPONENT Server)
-TARGET_LINK_LIBRARIES(mysql_upgrade mysqlclient)
+TARGET_LINK_LIBRARIES(mysql_upgrade mariadbclient)
 ADD_DEPENDENCIES(mysql_upgrade GenFixPrivs)
 
 MYSQL_ADD_EXECUTABLE(mysqlshow mysqlshow.c)
-TARGET_LINK_LIBRARIES(mysqlshow mysqlclient)
+TARGET_LINK_LIBRARIES(mysqlshow mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysql_plugin mysql_plugin.c)
-TARGET_LINK_LIBRARIES(mysql_plugin mysqlclient)
+TARGET_LINK_LIBRARIES(mysql_plugin mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysqlbinlog mysqlbinlog.cc)
-TARGET_LINK_LIBRARIES(mysqlbinlog mysqlclient)
+TARGET_LINK_LIBRARIES(mysqlbinlog mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysqladmin mysqladmin.cc)
-TARGET_LINK_LIBRARIES(mysqladmin mysqlclient)
+TARGET_LINK_LIBRARIES(mysqladmin mariadbclient)
 
 MYSQL_ADD_EXECUTABLE(mysqlslap mysqlslap.c)
 SET_SOURCE_FILES_PROPERTIES(mysqlslap.c PROPERTIES COMPILE_FLAGS "-DTHREADS")
-TARGET_LINK_LIBRARIES(mysqlslap mysqlclient)
+TARGET_LINK_LIBRARIES(mysqlslap mariadbclient)
 
 # "WIN32" also covers 64 bit. "echo" is used in some files below "mysql-test/".
 IF(WIN32)
@@ -81,7 +81,7 @@ ENDIF(WIN32)
 
 # async_example is just a code example, do not install it.
 ADD_EXECUTABLE(async_example async_example.c)
-TARGET_LINK_LIBRARIES(async_example mysqlclient)
+TARGET_LINK_LIBRARIES(async_example mariadbclient)
 
 SET_TARGET_PROPERTIES (mysqlcheck mysqldump mysqlimport mysql_upgrade mysqlshow mysqlslap mysql_plugin async_example
 PROPERTIES HAS_CXX TRUE)
--- mariadb-10.1.orig/libmysql/CMakeLists.txt
+++ mariadb-10.1/libmysql/CMakeLists.txt
@@ -278,7 +278,7 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
 
     INCLUDE (CheckCSourceCompiles)
     FILE(WRITE "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/src.ld"
-      "VERSION {\nlibmysqlclient_18 {\nglobal: *;\n};\n}\n")
+      "VERSION {\nlibmariadbclient_18 {\nglobal: *;\n};\n}\n")
     SET(CMAKE_REQUIRED_LIBRARIES "-Wl,src.ld")
     CHECK_C_SOURCE_COMPILES("int main() { return 0; }"
       SUPPORTS_VERSION_IN_LINK_SCRIPT)
@@ -380,18 +380,18 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
   # Generate version script.  
   # Create semicolon separated lists of functions to export from 
   # Since RPM packages use separate versioning for 5.1 API
-  # and 5.5 API (libmysqlclient_16 vs libmysqlclient_18), 
+  # and 5.5 API (libmariadbclient_16 vs libmariadbclient_18), 
   # we need 2 lists.
   SET (VERSION_HEADER
 "VERSION {
-  libmysqlclient_18 {
+  libmariadbclient_18 {
     global:")
   SET (VERSION_FOOTER
 "    local:
       *;
   };
 
-  libmysqlclient_16 {
+  libmariadbclient_16 {
     /* empty here. aliases are added above */
   };
 }
@@ -401,7 +401,7 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
   SET (CLIENT_API_5_1_ALIASES)
   FOREACH (f ${CLIENT_API_FUNCTIONS_5_1} ${CLIENT_API_5_1_EXTRA})
     SET(CLIENT_API_5_1_LIST "${CLIENT_API_5_1_LIST}\t${f};\n")
-    SET(CLIENT_API_5_1_ALIASES "${CLIENT_API_5_1_ALIASES}\"${f}@libmysqlclient_16\" = ${f};\n")
+    SET(CLIENT_API_5_1_ALIASES "${CLIENT_API_5_1_ALIASES}\"${f}@libmariadbclient_16\" = ${f};\n")
   ENDFOREACH()
       
   SET (CLIENT_API_5_5_LIST)
@@ -446,13 +446,13 @@ ADD_DEPENDENCIES(clientlib GenError)
 
 SET(LIBS clientlib dbug strings vio mysys mysys_ssl ${ZLIB_LIBRARY} ${SSL_LIBRARIES} ${LIBDL})
 
-# Merge several convenience libraries into one big mysqlclient
+# Merge several convenience libraries into one big mariadbclient
 # and link them together into shared library.
-MERGE_LIBRARIES(mysqlclient STATIC ${LIBS} COMPONENT Development)
+MERGE_LIBRARIES(mariadbclient STATIC ${LIBS} COMPONENT Development)
 
 # Visual Studio users need debug  static library for debug projects
 IF(MSVC)
- INSTALL_DEBUG_TARGET(mysqlclient DESTINATION ${INSTALL_LIBDIR}/debug)
+ INSTALL_DEBUG_TARGET(mariadbclient DESTINATION ${INSTALL_LIBDIR}/debug)
  INSTALL_DEBUG_TARGET(clientlib DESTINATION ${INSTALL_LIBDIR}/debug)
 ENDIF()
 
@@ -468,7 +468,7 @@ IF(UNIX)
       SET(${OUTNAME} ${LIBNAME}${EXTENSION}${DOT_VERSION})
     ENDIF() 
   ENDMACRO()
-  INSTALL_SYMLINK(${CMAKE_STATIC_LIBRARY_PREFIX}mysqlclient_r.a mysqlclient ${INSTALL_LIBDIR} Development)
+  INSTALL_SYMLINK(${CMAKE_STATIC_LIBRARY_PREFIX}mariadbclient_r.a mariadbclient ${INSTALL_LIBDIR} Development)
 ENDIF()
 
 IF(NOT DISABLE_SHARED)
@@ -484,9 +484,9 @@ IF(NOT DISABLE_SHARED)
     ELSE()
       SET(OS_SHARED_LIB_VERSION "${SHARED_LIB_MAJOR_VERSION}.0.0")
     ENDIF()
-    # Name of shared library is mysqlclient on Unix
+    # Name of shared library is mariadbclient on Unix
     SET_TARGET_PROPERTIES(libmysql PROPERTIES 
-      OUTPUT_NAME mysqlclient 
+      OUTPUT_NAME mariadbclient 
       VERSION "${OS_SHARED_LIB_VERSION}" 
       SOVERSION "${SHARED_LIB_MAJOR_VERSION}")
     IF(LINK_FLAG_NO_UNDEFINED OR VERSION_SCRIPT_LINK_FLAGS)
@@ -499,13 +499,13 @@ IF(NOT DISABLE_SHARED)
         "${libmysql_link_flags} ${LINK_FLAG_NO_UNDEFINED} ${VERSION_SCRIPT_LINK_FLAGS}")
     ENDIF() 
     # clean direct output needs to be set several targets have the same name
-    #(mysqlclient in this case)
-    SET_TARGET_PROPERTIES(mysqlclient PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+    #(mariadbclient in this case)
+    SET_TARGET_PROPERTIES(mariadbclient PROPERTIES CLEAN_DIRECT_OUTPUT 1)
     SET_TARGET_PROPERTIES(libmysql PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 
-    # Install links to libmysqlclient.so (client_r)
+    # Install links to libmariadbclient.so (client_r)
     GET_VERSIONED_LIBNAME(
-      "${CMAKE_SHARED_LIBRARY_PREFIX}mysqlclient_r"
+      "${CMAKE_SHARED_LIBRARY_PREFIX}mariadbclient_r"
       "${CMAKE_SHARED_LIBRARY_SUFFIX}"
       ""
       linkname)
@@ -514,7 +514,7 @@ IF(NOT DISABLE_SHARED)
     LIST(REMOVE_DUPLICATES OS_SHARED_LIB_SYMLINKS)
     FOREACH(ver ${OS_SHARED_LIB_SYMLINKS})
       GET_VERSIONED_LIBNAME(
-        "${CMAKE_SHARED_LIBRARY_PREFIX}mysqlclient_r"
+        "${CMAKE_SHARED_LIBRARY_PREFIX}mariadbclient_r"
         "${CMAKE_SHARED_LIBRARY_SUFFIX}"
         "${ver}"
         linkname)
--- mariadb-10.1.orig/man/mysql_config.1
+++ mariadb-10.1/man/mysql_config.1
@@ -38,7 +38,7 @@ supports the following options\&.
 \fB\-\-cflags\fR
 .sp
 Compiler flags to find include files and critical compiler flags and defines used when compiling the
-libmysqlclient
+libmariadbclient
 library\&. The options returned are tied to the specific compiler that was used when the library was created and might clash with the settings for your own compiler\&. Use
 \fB\-\-include\fR
 for more portable options that contain only include paths\&.
@@ -196,9 +196,9 @@ Usage: /usr/local/mysql/bin/mysql_config
 Options:
   \-\-cflags         [\-I/usr/local/mysql/include/mysql \-mcpu=pentiumpro]
   \-\-include        [\-I/usr/local/mysql/include/mysql]
-  \-\-libs           [\-L/usr/local/mysql/lib/mysql \-lmysqlclient \-lz
+  \-\-libs           [\-L/usr/local/mysql/lib/mysql \-lmariadbclient \-lz
                     \-lcrypt \-lnsl \-lm \-L/usr/lib \-lssl \-lcrypto]
-  \-\-libs_r         [\-L/usr/local/mysql/lib/mysql \-lmysqlclient_r
+  \-\-libs_r         [\-L/usr/local/mysql/lib/mysql \-lmariadbclient_r
                     \-lpthread \-lz \-lcrypt \-lnsl \-lm \-lpthread]
   \-\-socket         [/tmp/mysql\&.sock]
   \-\-port           [3306]
--- mariadb-10.1.orig/cmake/for_clients.cmake
+++ mariadb-10.1/cmake/for_clients.cmake
@@ -2,8 +2,8 @@
 # Generate LIBS and CFLAGS that third-party clients should use
 #
 
-# Use cmake variables to inspect dependencies for 
-# mysqlclient library (add -l stuff)
+# Use cmake variables to inspect dependencies for
+# mariadbclient library (add -l stuff)
 SET(CLIENT_LIBS "")
 SET(LIBS "")
 
@@ -45,10 +45,10 @@ MACRO(EXTRACT_LINK_LIBRARIES target var)
   ENDIF()
 ENDMACRO()
 
-EXTRACT_LINK_LIBRARIES(mysqlclient LIBS)
+EXTRACT_LINK_LIBRARIES(mariadbclient LIBS)
 EXTRACT_LINK_LIBRARIES(mysqlserver EMB_LIBS)
 
-SET(LIBS     "-lmysqlclient ${ZLIB_DEPS} ${LIBS} ${openssl_libs}")
+SET(LIBS     "-lmariadbclient ${ZLIB_DEPS} ${LIBS} ${openssl_libs}")
 SET(EMB_LIBS "-lmysqld ${ZLIB_DEPS} ${EMB_LIBS} ${openssl_libs}")
 
 MACRO(REPLACE_FOR_CLIENTS VAR)
--- mariadb-10.1.orig/scripts/make_win_bin_dist
+++ mariadb-10.1/scripts/make_win_bin_dist
@@ -293,7 +293,7 @@ mkdir -p $DESTDIR/lib/plugin
 cp sql/$TARGET/mysqld.lib $DESTDIR/lib/
 cp libmysql/$TARGET/libmysql.dll \
    libmysql/$TARGET/libmysql.lib \
-   libmysql/$TARGET/mysqlclient.lib \
+   libmysql/$TARGET/mariadbclient.lib \
    mysys/$TARGET/mysys.lib \
    regex/$TARGET/regex.lib \
    strings/$TARGET/strings.lib \
@@ -302,7 +302,7 @@ cp storage/*/$TARGET/ha_*.dll $DESTDIR/l
 
 if [ x"$TARGET" != x"release" ] ; then
   cp libmysql/$TARGET/libmysql.pdb \
-     libmysql/$TARGET/mysqlclient.pdb \
+     libmysql/$TARGET/mariadbclient.pdb \
      mysys/$TARGET/mysys.pdb \
      regex/$TARGET/regex.pdb \
      strings/$TARGET/strings.pdb \
@@ -318,8 +318,8 @@ if [ x"$PACK_DEBUG" = x"" -a -f "libmysq
   cp libmysql/debug/libmysql.dll \
      libmysql/debug/libmysql.lib \
      libmysql/debug/libmysql.pdb \
-     libmysql/debug/mysqlclient.lib \
-     libmysql/debug/mysqlclient.pdb \
+     libmysql/debug/mariadbclient.lib \
+     libmysql/debug/mariadbclient.pdb \
      mysys/debug/mysys.lib \
      mysys/debug/mysys.pdb \
      regex/debug/regex.lib \
--- mariadb-10.1.orig/scripts/mysql_config.pl.in
+++ mariadb-10.1/scripts/mysql_config.pl.in
@@ -183,15 +183,15 @@ if ( $^O eq "MSWin32" )
   my $linkpath   = "$pkglibdir";
   # user32 is only needed for debug or embedded
   my @winlibs = ("wsock32.lib","advapi32.lib","user32.lib");
-  @lib_opts   = ("$linkpath/mysqlclient.lib",@winlibs);
+  @lib_opts   = ("$linkpath/mariadbclient.lib",@winlibs);
   @lib_r_opts = @lib_opts;
   @lib_e_opts = ("$linkpath/mysqlserver.lib",@winlibs);
 }
 else
 {
   my $linkpath   = "-L$pkglibdir @RPATH_OPTION@";
-  @lib_opts   = ($linkpath,"-lmysqlclient");
-  @lib_r_opts = ($linkpath,"-lmysqlclient_r");
+  @lib_opts   = ($linkpath,"-lmariadbclient");
+  @lib_r_opts = ($linkpath,"-lmariadbclient_r");
   @lib_e_opts = ($linkpath,"-lmysqld");
 }
 
--- mariadb-10.1.orig/tests/CMakeLists.txt
+++ mariadb-10.1/tests/CMakeLists.txt
@@ -18,12 +18,12 @@ ADD_DEFINITIONS("-DMYSQL_CLIENT")
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
 
 ADD_EXECUTABLE(mysql_client_test mysql_client_test.c)
-TARGET_LINK_LIBRARIES(mysql_client_test mysqlclient)
+TARGET_LINK_LIBRARIES(mysql_client_test mariadbclient)
 SET_TARGET_PROPERTIES(mysql_client_test PROPERTIES LINKER_LANGUAGE CXX)
 
 IF(WITH_UNIT_TESTS)
   ADD_EXECUTABLE(bug25714 bug25714.c)
-  TARGET_LINK_LIBRARIES(bug25714 mysqlclient)
+  TARGET_LINK_LIBRARIES(bug25714 mariadbclient)
   SET_TARGET_PROPERTIES(bug25714 PROPERTIES LINKER_LANGUAGE CXX)
 ENDIF()
 
@@ -34,7 +34,7 @@ FIND_LIBRARY(EVENT_LIBRARY event)
 MARK_AS_ADVANCED(EVENT_LIBRARY)
 IF(HAVE_EVENT_H AND EVENT_LIBRARY)
   ADD_EXECUTABLE(async_queries async_queries.c)
-  TARGET_LINK_LIBRARIES(async_queries mysqlclient ${EVENT_LIBRARY})
+  TARGET_LINK_LIBRARIES(async_queries mariadbclient ${EVENT_LIBRARY})
   SET_TARGET_PROPERTIES(async_queries PROPERTIES LINKER_LANGUAGE CXX)
 
 ENDIF()
--- mariadb-10.1.orig/unittest/mysys/CMakeLists.txt
+++ mariadb-10.1/unittest/mysys/CMakeLists.txt
@@ -21,7 +21,7 @@ MY_ADD_TESTS(my_vsnprintf LINK_LIBRARIES
 ADD_DEFINITIONS(${SSL_DEFINES})
 
 MY_ADD_TESTS(ma_dyncol
-	     LINK_LIBRARIES mysqlclient)
+	     LINK_LIBRARIES mariadbclient)
 
 IF(WIN32)
   MY_ADD_TESTS(my_delete LINK_LIBRARIES mysys)
